Mathlib/Data/ZMod/Coprime.lean

```diff
@@ -37,4 +37,114 @@ theorem eq_zero_of_gcd_ne_one {a : ℤ} {p : ℕ} (pp : p.Prime) (h : a.gcd p 
   (@eq_zero_iff_gcd_ne_one a p ⟨pp⟩).mpr h
 #align zmod.eq_zero_of_gcd_ne_one ZMod.eq_zero_of_gcd_ne_one
 
+open Nat
+
+lemma coprime_list_prod_iff_right {k} {l : List ℕ} :
+    Coprime k l.prod ↔ ∀ n ∈ l, Coprime k n := by
+  induction' l with m l ih <;> simp[Nat.coprime_mul_iff_right, *]
+
+lemma coprime_list_prod_iff_left {k} {l : List ℕ} :
+    Coprime l.prod k ↔ ∀ n ∈ l, Coprime k n := by
+  induction' l with m l ih <;> simp[Nat.coprime_mul_iff_left, *]
+  intro
+  exact coprime_comm
+
+lemma pairwise_coprime_cons_iff_pairwise_coprime_coprime_prod {n} {l : List ℕ} :
+    (n :: l).Pairwise Coprime ↔ l.Pairwise Coprime ∧ Coprime n l.prod := by
+  rw[List.pairwise_cons, coprime_list_prod_iff_right, and_comm]
+
+lemma modEq_iff_modEq_list_prod {a b} {l : List ℕ} (co : l.Pairwise Coprime) :
+    (∀ i, a ≡ b [MOD l.get i]) ↔ a ≡ b [MOD l.prod] := by
+  induction' l with m l ih <;> simp[Nat.modEq_one]
+  · rcases co with (_ | ⟨hm, hl⟩)
+    have : a ≡ b [MOD m] ∧ a ≡ b [MOD l.prod] ↔ a ≡ b [MOD m * l.prod] :=
+      Nat.modEq_and_modEq_iff_modEq_mul (coprime_list_prod_iff_right.mpr hm)
+    simp[← this, ← ih hl]
+    constructor
+    · intro h; exact ⟨by simpa using h ⟨0, by simp⟩, fun i => by simpa using h i.succ⟩
+    · intro h i
+      cases i using Fin.cases
+      · simpa using h.1
+      · simpa using h.2 _
+
+/-- List of coprimes, using the Chinese remainder theorem -/
+def chineseRemainderList : (l : List (ℕ × ℕ)) → (H : (l.map Prod.snd).Pairwise Coprime) →
+    { k // ∀ i, k ≡ (l.get i).1 [MOD (l.get i).2] }
```

Comment:
Please explain in the docstring what `chineseRemainderList l hl` means; your current description doesn't mention `l` at all.

Reply:
Further more, this name is bizarre because the return value isn't a list at all, so this certainly is not a "list of coprimes"!

Reply:
Renamed to `chineseRemainderOfList`. Also added an explanation to make it clear that it is a list version of `chineseRemainder`.